### PR TITLE
API reference: Shorten supported argument types in docstrings

### LINKED ISSUE #2718: Shorten the list of supported argument types in docstrings
I'll take the `spacing` parameter as an example, which is used in the [`xyz2grd`](https://www.pygmt.org/dev/api/generated/pygmt.xyz2grd.html) and many other functions.

Here is the source code of the `spacing` docstrings:
https://github.com/GenericMappingTools/pygmt/blob/5b4d5c37b9196299b5ff43fd07aff30f9d654211/pygmt/helpers/decorators.py#L300-L302

Here is what it looks like:
image

As you can see, currently, `spacing` is documented to support 5 types: int, float, str, list and tuple.

My thoughts:

1. Simplify `int or float` to `float`. If an argument can accept float (e.g., 1.0), it definitely can accept int (e.g., 1). It's not necessary to list both. 
2. Simplify `list or tuple` to `list` or `tuple` or something else.

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PRs aims to shorten the supported argument types in docstrings:
- Change `int or float` to `float`
- Change `list or tuple` to `list`
- Change `type1 or typ2 or typ3` to `type1, typ2, or typ3`

Fixes #2718

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
Need to fix line 419 in `pygmt/helpers/decorators.py`:
```
=================================== FAILURES ===================================
_______________ [doctest] pygmt.helpers.decorators.fmt_docstring _______________
404     ...         Pass in either a file name to an ASCII data table, a 2-D
405     ...         {table-classes}.
406     ...     {region}
407     ...     {projection}
408     ...
409     ...     {aliases}
410     ...     '''
411     ...     pass
412     ...
413     >>> print(gmtinfo.__doc__)
Differences (unified diff with -expected +actual):
    @@ -4,5 +4,5 @@
     Parameters
     ----------
    -data : str or numpy.ndarray or pandas.DataFrame or xarray.Dataset or geo...
    +data : str or numpy.ndarray, pandas.DataFrame, xarray.Dataset, or geopandas.GeoDataFrame
         Pass in either a file name to an ASCII data table, a 2-D
         :class:`numpy.ndarray`, a :class:`pandas.DataFrame`, an
```

**Comment by yvonnefroehlich** [comment]
> Need to fix line 419 in `pygmt/helpers/decorators.py`:

This is updated in the commits https://github.com/GenericMappingTools/pygmt/pull/2746/commits/213d4b7912613ffa2e8ee2b6230ae917170fba58 and https://github.com/GenericMappingTools/pygmt/pull/2746/commits/1e3abc9be23ce616813fb81e9a6d0a417f422cb2.

**Review by seisman — APPROVED** [review]

File: pygmt/helpers/decorators.py [hunk 1]
```diff
@@ -416,7 +416,7 @@ def fmt_docstring(module_func):
     <BLANKLINE>
     Parameters
     ----------
-    data : str or numpy.ndarray or pandas.DataFrame or xarray.Dataset or geo...
+    data : str, numpy.ndarray, pandas.DataFrame, xarray.Dataset, or geo...
```
Comment (seisman) [inline pygmt/helpers/decorators.py:419]: Line 403, need to change `str or {table-like}` to `str, {table-like}`

Comment (yvonnefroehlich) [reply pygmt/helpers/decorators.py:419]: This is updated in commit https://github.com/GenericMappingTools/pygmt/pull/2746/commits/335049ac613dc1e8210bafeba73ccee698178293.

**Pull request merged**